mithril-common/src/chain_observer/model.rs

```diff
@@ -1,12 +1,34 @@
 use anyhow::anyhow;
-use serde::Serialize;
+// required for derive attrs to work
+use pallas_codec::utils::{Int, KeyValuePairs};
+use serde::{Deserialize, Serialize};
 use serde_json::Value;
 use std::collections::HashMap;
 use strum::{Display, EnumDiscriminants};
 use thiserror::Error;
 
 use crate::{StdError, StdResult};
 
+#[derive(Deserialize, Debug, PartialEq, Eq, PartialOrd, Ord, Clone)]
+pub struct Constr<A> {
+    pub constructor: Option<u64>,
+    pub fields: Vec<A>,
+}
+
+#[derive(Deserialize, Debug, PartialEq, Eq, PartialOrd, Ord, Clone)]
+#[serde(rename_all = "lowercase")]
+pub enum Metadatum {
+    Datum(Constr<Metadatum>),
+    Int(Int),
+    Bytes(String),
+    Text(String),
+    List(Vec<Metadatum>),
+    Map(KeyValuePairs<Metadatum, Metadatum>),
+}
```

Comment:
there's no need to replicate this, `pallas::primitives` already has all of these structures defined and with the corresponding CBOR codec.

Reply:
done.